### PR TITLE
refactor: Manage Config Controller to API Only

### PR DESCRIPTION
**Pull request by JeremyRudman**

### CODE REVIEW

File: app/controllers/manage/configs_controller.rb [hunk 1]
```diff
@@ -2,34 +2,24 @@ class Manage::ConfigsController < Manage::ApplicationController
...
 
   def update
     key = @config.var.to_sym
     value = params[:hackathon_config][key]
     value = true if value == "true"
     value = false if value == "false"
     if @config.var.start_with?("agreement_") && !value.start_with?('http://', 'https://')
-      flash[:alert] = "Config \"#{key}\" must start with http:// or https://"
-      render :edit
+      render ErrorResponse.new(:config_update_agreementMustStartHTTP), status: :unprocessable_entity
```
Comment (JeremyRudman) [inline app/controllers/manage/configs_controller.rb]: not sure if the new error message is any good

File: app/controllers/manage/configs_controller.rb [hunk 2]
```diff
@@ -2,34 +2,22 @@ class Manage::ConfigsController < Manage::ApplicationController
   before_action :require_director
   before_action :get_config, only: [:edit, :update, :update_only_css_variables]
 
-  respond_to :html, :json
-
-  def index
-    @config = HackathonConfig.get_all
```
Comment (cbaudouinjr) [inline app/controllers/manage/configs_controller.rb:8]: I think we still need the index so we can get all the configs.

File: test/controllers/manage/configs_controller_test.rb [hunk 3]
```diff
@@ -2,17 +2,7 @@
 
 class Manage::ConfigsControllerTest < ActionController::TestCase
   context "while not authenticated" do
-    should "redirect to sign in page on manage_configs#index" do
-      get :index
```
Comment (cbaudouinjr) [inline test/controllers/manage/configs_controller_test.rb]: This will need to be re-added if we need the index.

**Review by cbaudouinjr — CHANGES_REQUESTED** [review]

File: app/controllers/manage/configs_controller.rb (hunk 2 above)
Comment (JeremyRudman) [reply app/controllers/manage/configs_controller.rb:8]: re added index

**Review by cbaudouinjr — APPROVED** [review]

**Pull request closed without merging**